File: Client/Clan.js
export class Clan{

    constructor(id, ime, prezime, brojDozvole, tipDozvole){
        this.id=id;
        this.ime=ime;
        this.prezime=prezime;
        this.brojDozvole=brojDozvole;
        this.tipDozvole=tipDozvole;
    }

    crtaj(host){
        var tr=document.createElement("tr");
        host.appendChild(tr);

        var el=document.createElement("td");
        el.innerHTML=this.ime;
        tr.appendChild(el);

        el=document.createElement("td");
        el.innerHTML=this.prezime;
        tr.appendChild(el);

        el=document.createElement("td");
        el.innerHTML=this.brojDozvole;
        tr.appendChild(el);

        el=document.createElement("td");
        el.innerHTML=this.tipDozvole;
        tr.appendChild(el);
    }
}

File: Models/Clan.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace Models
{
    [Table("Clan")]
    public class Clan
    {
        [Key]
        public int ID { get; set; }
        
        [Required]
        [MaxLength(50)]
        public string Ime { get; set; }

        [Required]
        [MaxLength(50)]
        public string Prezime { get; set; }
        
        [Range(100,999)]
        public int? BrojDozvole { get; set; }
        
        [MaxLength(50)]
        public string TipDozvole { get; set; }

        [JsonIgnore]
        public List<Aktivnost> Aktivnosti { get; set; }

        public static explicit operator Clan(Aktivnost v)
        {
            throw new NotImplementedException();
        }
    }
}